### DISCUSSION TITLE
Cannot run example `panda_pick_and_place.py` on Ubuntu Bionic

### DISCUSSION THREAD

**Discussion by nicholaspalomo** [robotology/gym-ignition#351] [General]
I cannot run the `panda_pick_and_place.py` example.

When running:
``` python examples/panda_pick_and_place.py ```

I get the following traceback:
```
Traceback (most recent call last):
  File "examples/panda_pick_and_place.py", line 8, in <module>
    from gym_ignition.rbd import conversions
ModuleNotFoundError: No module named 'gym_ignition.rbd'
```

To reproduce:
- I went through the setup/installation instructions here: https://github.com/robotology/gym-ignition
- OS: Ubuntu 18.04.
- Python version: Python 3.6.9 (also tried with Python 3.8.10).
- I can run the cartpole example just fine.

Thanks for all your work on this project. It looks really cool!

**Comment by diegoferigo** [comment]
Hi @nicholaspalomo, thanks for opening this issue. Your Ubuntu version is not currently supported, as reported in our [Support Policy](https://robotology.github.io/gym-ignition/master/installation/support_policy.html).

Regardless, can you try to provide the output of the following commands:

```python
>>> import gym_ignition
```

```python
>>> from gym_ignition.rbd import conversions
```

```bash
sudo apt-get install tree
tree $(python -c "import gym_ignition, pathlib; print(pathlib.Path(gym_ignition.__file__).parent)")
```

Edit: moved to discussions since it is likely related to system configuration rather than code bug.

**Reply by nicholaspalomo** [reply]
Hi, @diegoferigo. Thanks for your answer.

With the small changes in this commit, I was able to get the Panda pick-and-place example to run!
https://github.com/nicholaspalomo/gym-ignition/commit/95bc3b1da92fe1e805c571223e39e3c07d5f9d07

I also had to copy some of the modules from the `python` directory of the repo to the `site-packages` folder of my virtual environment.

I'm looking forward to using gym-ignition. I'd like to try adding Spot to it and developing a locomotion controller for the robot. Currently I use raisim (https://raisim.com/) from ETH Zurich for RL. While being very good, however, it is not open source.

**Reply by diegoferigo** [reply]
I'm glad you managed to make the project run on your setup! 

> I also had to copy some of the modules from the python directory of the repo to the site-packages folder of my virtual environment.

Mmh should not be necessary. From a cloned repository, the normal installation (`pip install .`) should build the wheel and install the whole Python package and the C++ bindings into your site-packages folder. Instead, an editable installation (`pip install -e .`) would just install the Python package without the C++ part. I'm not yet sure what could be the source of the original problem regarding `gym_ignition.rbd` from the data you provided.

Few comments on https://github.com/nicholaspalomo/gym-ignition/commit/95bc3b1da92fe1e805c571223e39e3c07d5f9d07:

- Instead of `len(panda.joint_names())`, you can use `panda.dofs()`. Even though at the moment we only support joints with just 1 DoF, in the future there could be multi-dofs joints (e.g. ball). In that case, the two methods would return different numbers, and `Model.dofs()` always yields the correct number.
- I'm not sure why the removed `*.to_gazebo()` is necessary to run the code. To be honest, that's a leftover, they shouldn't be there in the first place since the `Link.enable_contact_detection` method is part of the core ScenarIO APIs and is not an implementation detail of the Gazebo backend (which methods are exposed through `to_gazebo()`). However, it should not make any difference. Regardless, I will update the example code.

For those reading this discussion in the future, I also link https://github.com/robotology/gym-ignition/issues/288#issuecomment-793997038 in which some more details about the Ubuntu Bionic support was discussed. I update the title accordingly.

**Reply by diegoferigo** [reply]
> I'm looking forward to using gym-ignition. I'd like to try adding Spot to it and developing a locomotion controller for the robot. Currently I use raisim (https://raisim.com/) from ETH Zurich for RL. While being very good, however, it is not open source.

Cool! When you have something working, feel free to provide pointers in a new [Show and tell](https://github.com/robotology/gym-ignition/discussions/categories/show-and-tell) discussion if you feel like. I've played around a bit with raisim in the past and I have to say that it's really a good piece of software. I have to tell you that Ignition Gazebo with DART has not yet that level of performance, raisim is much much faster. Though, there are trade offs, as you said it's closed source and in research I believe that being able to hack and improve the software we use is paramount. If we need a new feature, we need to be able implementing it even if it sits in our forks forever, and this is not possible with closed source software. Furthermore, I'm not sure how well the raisim license behaves if you want to deploy the RL environment in an external cluster (similarly to mujoco).
